cmd/object-handlers.go

```diff
@@ -2129,6 +2131,36 @@ func (api objectAPIHandlers) ListObjectPartsHandler(w http.ResponseWriter, r *ht
 	writeSuccessResponseXML(w, encodedSuccessResponse)
 }
 
+// Send white spaces every 10 seconds to the client till completeMultiPartUpload() is done so that the client does not time out.
+// Downside is we might send 200OK and then send error XML. But accoording to S3 spec
+// the client is supposed to check for error XML even if it received 200OK. But for erasure this is not a
+// problem as completeMultiPartUpload() is quick. Even For FS, it would not be an issue
+// as we do background append as and when the parts arrive and completeMultiPartUpload is quick.
+// Only in a rare case where parts would be out of order will FS' completeMultiPartUpload() take a longer time.
+func sendWhiteSpace(ctx context.Context, w http.ResponseWriter) <-chan struct{} {
+	doneCh := make(chan struct{})
+	go func() {
+		ticker := time.NewTicker(time.Second * 10)
+		for {
+			select {
+			case <-ticker.C:
+				// Write a white space char to the client to prevent client timeouts
+				// when the server takes a long time to completeMultiPartUpload()
+				if _, err := w.Write([]byte(" ")); err != nil {
```

Comment:
For whitespace perhaps we can do HTTP style.  `w.Write([]byte("\n\r"))` , all clients skip this properly..